src/stability_sdk/client.py

```diff
@@ -265,38 +275,65 @@ def generate(
         else:
             seed = list(seed)
 
-        prompt_ = []
-        if isinstance(prompt, str):
-            prompt_ = [generation.Prompt(text=prompt)]
-        elif isinstance(prompt, Sequence):
-            prompt_ = [generation.Prompt(text=p) for p in prompt]
-        else:
-            raise TypeError("prompt must be a string or a sequence")
-
+        prompt_: List[generation.Prompt] = []
+        if any(isinstance(prompt, t) for t in (str, generation.Prompt)):
+            prompt = [prompt]
+        for p in prompt:
+            if isinstance(p, str):
+                p = generation.Prompt(text=p)
+            elif not isinstance(p, generation.Prompt):
+                raise TypeError("prompt must be a string or generation.Prompt object")
+            prompt_.append(p)
+
+        step_parameters = dict(
+            scaled_step=0,
+            sampler=generation.SamplerParameters(cfg_scale=cfg_scale),
+            schedule=generation.ScheduleParameters(
+                start=start_schedule,
+                end=end_schedule,
+            ),
+        )
+            
         if init_image is not None:
             prompt_ += [image_to_prompt(init_image, init=True)]
-            parameters = (
-                generation.StepParameter(
-                    scaled_step=0,
-                    sampler=generation.SamplerParameters(
-                        cfg_scale=cfg_scale,
-                    ),
-                    schedule=generation.ScheduleParameters(
-                        start=start_schedule,
-                        end=end_schedule,
-                    ),
-                ),
-            )
+
             if mask_image is not None:
                 prompt_ += [image_to_prompt(mask_image, mask=True)]
-        else:
-            parameters = (
-                generation.StepParameter(
-                    scaled_step=0,
-                    sampler=generation.SamplerParameters(cfg_scale=cfg_scale),
-                ),
-            )
 
+        
+        if guidance_prompt:
+            if isinstance(guidance_prompt, str):
+                guidance_prompt = generation.Prompt(text=guidance_prompt)
+            elif not isinstance(guidance_prompt, generation.Prompt):
+                raise ValueError("guidance_prompt must be a string or Prompt object")
+        if guidance_strength == 0.0:
+            guidance_strength = None
+
+            
+        # Build our CLIP parameters
+        if guidance_preset is not generation.GUIDANCE_PRESET_NONE:
```

Comment:
If setting guidance, we need to set the diffuser to `SAMPLER_K_DPM_2_ANCESTRAL`, unless specifically set by the user. 

But right now, we could solve this by setting the diffuser to `None`,  and letting the server set it?